### PR TITLE
chore: enhance types for base-command.ts

### PR DESCRIPTION
**Pull request by lukasholzer**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

To not have my change and the types refactoring in one PR this is just the PR for the types refactorings.

The actual change will come in a follow-up.

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/commands/base-command.ts [hunk 1]
```diff
@@ -229,38 +203,28 @@ export default class BaseCommand extends Command {
...
-      await this.init(actionCommand)
+      await this.init(actionCommand as BaseCommand)
       debug(`${name}:preAction`)('end')
     })
   }
 
-  /** @private */
-  noBaseOptions = false
-
+  #noBaseOptions = false
```
Comment (eduardoboucas) [inline src/commands/base-command.ts:211]: Can we not use `private noBaseOptions`?

File: src/commands/base-command.ts [hunk 2]
```diff
@@ -613,7 +548,8 @@ export default class BaseCommand extends Command {
       certificateFile: flags.httpProxyCertificateFilename,
     })
     const apiOpts = { ...apiUrlOpts, agent }
-    const api = new NetlifyAPI(token || '', apiOpts)
+    // TODO: remove typecast once we have proper types for the API
```
Comment (eduardoboucas) [inline src/commands/base-command.ts:551]: 🙏🏻

File: src/utils/init/utils.ts [hunk 3]
```diff
@@ -1,53 +1,57 @@
...
+ * there our build will automatically install the latest without
+ * user management of the versioning.
+ * @param pluginsInstalled
+ * @param pluginsRecommended
  * @returns
  */
-export const getPluginsToAutoInstall = (pluginsInstalled = [], pluginsRecommended = []) =>
-  pluginsRecommended.reduce(
+export const getPluginsToAutoInstall = (
+  command: BaseCommand,
```
Comment (Skn0tt) [inline src/utils/init/utils.ts:32]: I don't see this parameter used anywhere - am I missing something?

**Review by eduardoboucas — APPROVED** [review]

File: src/commands/base-command.ts (hunk 1 above)
Comment (lukasholzer) [reply src/commands/base-command.ts:211]: TypeScript private is not a JavaScript private this is truly private and not accessible after transpiling.

TBH I would prefer this notation over using they private keyword as this is just dropped on transpiling

File: src/utils/init/utils.ts (hunk 3 above)
Comment (lukasholzer) [reply src/utils/init/utils.ts:32]: Nope it is a prerequisite for the follow up PR just forgot to remove it from this one. In the follow up I need to access the feature flags from the base command.

File: src/commands/base-command.ts (hunk 1 above)
Comment (eduardoboucas) [reply src/commands/base-command.ts:211]: Do we care if it's truly private or not, though? I think we should focus on code readability and type safety, and having to prefix every private property with `#` feels a bit awkward and noisy to me, personally.

**Pull request merged**